src/main/java/picard/analysis/StrandSpecificityUtil.java

```diff
@@ -0,0 +1,196 @@
+                final LocusFunction[] locusFunctions = new LocusFunction[alignmentBlock.getLength()];
+
+                // By default, if base does not overlap with rRNA or gene, it is intergenic.
+                Arrays.fill(locusFunctions, 0, locusFunctions.length, LocusFunction.INTERGENIC);
+
+                // Assign the function for each transcripts
+                for (final Gene gene : overlappingGenes) {
+                    for (final Gene.Transcript transcript : gene) {
+                        transcript.assignLocusFunctionForRange(alignmentBlock.getReferenceStart(), locusFunctions);
+                    }
+                }
+
+                // Tally the function of each base in the alignment block.
+                for (final LocusFunction locusFunction : locusFunctions) {
+                    switch (locusFunction) {
+                        case UTR:
+                        case CODING:
+                            return true;
+                        case INTERGENIC:
+                        case INTRONIC:
+                        case RIBOSOMAL:
+                            break;
+                    }
+                }
+
+                return false;
+            });
+
+            // Only examine reads that overlap an exon
+            if (overlapsExon) {
+                // We would need both read pairs to check multiple genes, and therefore we ignore such cases and call
+                // them unclassified later.
+                if (overlappingGenes.stream().map(Gene::isPositiveStrand).distinct().count() == 1) {
+                    final boolean negativeReadStrand            = rec.getReadNegativeStrandFlag();
+                    final boolean negativeTranscriptionStrand   = overlappingGenes.iterator().next().isNegativeStrand();
+
+                    if (!rec.getReadPairedFlag() || rec.getFirstOfPairFlag()) {
+                        if (negativeReadStrand == negativeTranscriptionStrand) readOneAgree    += 1;
+                        else                                                   readOneDisagree += 1;
+                    } else {
+                        if (negativeReadStrand == negativeTranscriptionStrand) readTwoAgree    += 1;
+                        else                                                   readTwoDisagree += 1;
+                    }
+                }
+                numReadsExamined += 1;
+            }
+        }
+
+        // NB: these two values may not add up to one if some reads cannot be classified due to overlapping multiple
+        // genes that have different transcription strands
+        double fractionFirstReadTranscriptionStrand  = (readOneAgree + readTwoDisagree) / (double) numReadsExamined;
+        double fractionSecondReadTranscriptionStrand = (readOneDisagree + readTwoAgree) / (double) numReadsExamined;
+        double fractionUnclassified = (numReadsExamined - readOneAgree - readOneDisagree - readTwoAgree - readTwoDisagree) / (double) numReadsExamined;
```

Comment:
Looking at the loop as it's presently constructed, I don't see a way for this to be anything other than 0 since you always add each read to one of the four counts.  What am I missing?